huggingface_pipelines/config.py

```diff
@@ -0,0 +1,196 @@
+from abc import ABC
+from dataclasses import dataclass, replace
+from typing import List, TypedDict
+
+
+class PipelineOverwrites(TypedDict, total=False):
+    batch_size: int
+    device: str
+    cache_to_arrow: bool
+    take: int
+    output_dir: str
+    output_file_name: str
+    columns: List[str]
+
+
+class DatasetOverwrites(TypedDict, total=False):
+    dataset_name: str
+    dataset_split: str
+    num_shards: int
+    shard_id: int
+    cache_dir: str
+
+
+class TextToEmbeddingOverwrites(PipelineOverwrites, total=False):
+    encoder_model: str
+    source_lang: str
+
+
+class EmbeddingToTextOverwrites(PipelineOverwrites, total=False):
+    decoder_model: str
+    target_lang: str
+
+
+class AudioOverwrites(PipelineOverwrites, total=False):
```

Comment:
I would go for a structure where each python module (like `huggingface_pipelines/text.py` or `huggingface_pipelines/audio.py`) has a corresponding config and pipelines definition inside and they are not written in a common module.